src/plugins_3rdparty/kalign/src/KalignPlugin.cpp

```diff
@@ -176,9 +177,18 @@ void KalignMSAEditorContext::sl_align() {
     assert(action != NULL);
     MSAEditor *ed = action->getMSAEditor();
     MultipleSequenceAlignmentObject *obj = ed->getMaObject();
-
     KalignTaskSettings s;
     QObjectScopedPointer<KalignDialogController> dlg = new KalignDialogController(ed->getWidget(), obj->getMultipleAlignment(), s);
+    bool containAsterisk = false;
+
+    for (const MultipleSequenceAlignmentRow &row : obj->getMsa()->getMsaRows()) {
```

Comment:
1) Use qAsConst with for loop & QT collections
2) Do not use collections returned by value in (see getMsaRows()) in for loop. It creates temporary object with an undefined life range. (qAsConst will help here too)